docs/pages/components/Panel.mdx

```diff
@@ -0,0 +1,261 @@
+</Note>
+
+```jsx example
+<Panel aria-label="Panel Heading">
+  <p>Panel Content</p>
+</Panel>
+```
+
+### Panel with Heading (Prop)
+
+To add a heading to a `Panel`, you can use the `heading` prop. One advantage to using this prop over composing
+the elements is it automatically adds an `id` to the heading and `aria-labelledby` to the `Panel` component.
+
+Inspect the below example and you will see that the `Panel` component has an `aria-labelledby` attribute with the 
+value of the `id` of the heading.
+
+```jsx example
+<Panel heading={{ id: 'panel-heading-prop-example', text: 'Panel Heading', level: 3 }}>
+  <p>Panel Content</p>
+</Panel>
+```
+
+### Panel (Composed)
+
+In addition to using the `heading` prop, you can also compose the `Panel` component with the `PanelHeader` and 
+`PanelContent` components.
+
+<Note>
+  The `PanelHeader` and `PanelContent` should not be used with the `Panel`
+  heading prop at the same time. Choose one or the other to use. 
+  
+  In addition, if you use the `PanelHeader`, you are responsible for adding `aria-labelledby` to
+  the `Panel` and an ID to the heading.
+</Note>
+
+```jsx example
+<Panel>
+  <PanelHeader>
+    <h3>Panel Header</h3>
+  </PanelHeader>
+  <PanelContent>
+    <p>Panel Content</p>
+  </PanelContent>
+</Panel>
+```
+
+### Disabling Panel Padding
+
+For greater control over the styling of the `Panel`'s content, you can disable the default padding by setting 
+the `padding` prop to `false`.
+
+```jsx example
+<Panel
+  heading={{ id: 'example', text: 'Panel Heading', level: 3 }}
+  padding={false}
+>
+  <p style={{ paddingLeft: '1rem' }}>Panel Content</p>
+</Panel>
+```
+
+### Disabling Panel Content Padding
+
+In addition to being able to disable the `Panel` padding via its `padding` prop, the `PanelContent` also has a `padding`
+prop which can be set to `false`. An example where this may be useful is when trying to display content in a two 
+column or grid layout. 
+
+In the following example, we are using CSS `columns` to create the two column layout. 
+
+```jsx example
+<Panel>
+  <PanelHeader
+    style={{
+      display: 'flex',
+      justifyContent: 'space-between',
+      alignItems: 'center',
+      flexWrap: 'wrap'
+    }}
+  >
+    <h3>Panel Header</h3>
+    <Button variant="primary" thin>
+      <Icon type="plus" />
+      Button
+    </Button>
+  </PanelHeader>
+  <PanelContent
+    padding={false}
+    style={{
+      columns: '2',
+      columnRule: 'var(--panel-divider-border)',
+      columnGap: '0'
+    }}
+  >
+    <div style={{ padding: 'var(--space-small)' }}>Column 1</div>
+    <div style={{ padding: 'var(--space-small)' }}>Column 2</div>
+  </PanelContent>
+  <PanelContent>
+    <p>Footer</p>
+  </PanelContent>
+</Panel>
+```
+
+### Header With Flexbox Content
```

Comment:
I'm not 100% sure this section is needed as we haven't really suggested this in other documentation as other components can render however you put stuff in them as well. Maybe it is useful here though since this is more of a container component. What do you think?

Reply:
Commented below, but think it would definitely be useful to include at least one example like this (either this one or the one below) since its used in a lot of the designs we have/is a pretty common use case generally. Also demonstrates some of the real usefulness of using composition over the heading prop, IMO.